src/RoomsModel.cpp

```diff
@@ -0,0 +1,64 @@
+#include "RoomsModel.h"
+
+#include "Cache_p.h"
+#include "CompletionModelRoles.h"
+
+RoomsModel::RoomsModel(bool showOnlyRoomWithAliases, QObject *parent)
+  : QAbstractListModel(parent)
+  , showOnlyRoomWithAliases_(showOnlyRoomWithAliases)
+{
+        std::vector<std::string> rooms_ = cache::joinedRooms();
+        roomInfos                       = cache::getRoomInfo(rooms_);
+
+        for (const auto &r : rooms_) {
+                auto roomAliasesList = cache::client()->getRoomAliases(r);
+
+                if (showOnlyRoomWithAliases_) {
+                        if (roomAliasesList) {
+                                roomids.push_back(QString::fromStdString(r));
+                                roomAliases.push_back(
+                                  QString::fromStdString(roomAliasesList->alias));
+                        }
+                } else {
+                        roomids.push_back(QString::fromStdString(r));
+                        roomAliases.push_back(
+                          roomAliasesList ? QString::fromStdString(roomAliasesList->alias) : "");
+                }
+        }
+}
+
+QHash<int, QByteArray>
+RoomsModel::roleNames() const
+{
+        return {{CompletionModel::CompletionRole, "completionRole"},
+                {CompletionModel::SearchRole, "searchRole"},
+                {CompletionModel::SearchRole2, "searchRole2"},
+                {Roles::RoomAlias, "roomAlias"},
+                {Roles::AvatarUrl, "avatarUrl"},
+                {Roles::RoomID, "roomid"},
+                {Roles::RoomName, "roomName"}};
+}
+
+QVariant
+RoomsModel::data(const QModelIndex &index, int role) const
+{
+        if (hasIndex(index.row(), index.column(), index.parent())) {
+                switch (role) {
+                case CompletionModel::CompletionRole:
+                        return QString("[%1](https://matrix.to/%1)").arg(roomAliases[index.row()]);
```

Comment:
```suggestion
                        return QString("[%1](https://matrix.to/#/%1)").arg(QUrl::toPercentEncoding(roomAliases[index.row()]));
```

See https://matrix.org/docs/spec/appendices#matrix-to-navigation